pinot-spi/src/main/java/org/apache/pinot/spi/config/table/assignment/SegmentAssignmentConfig.java

```diff
@@ -34,6 +34,7 @@ public SegmentAssignmentConfig(@JsonProperty(value = "segmentAssignmentStrategy"
     _assignmentStrategy = assignmentStrategy;
   }
 
+  @JsonProperty("segmentAssignmentStrategy")
```

Comment:
Hi @gortiz , thank for your reply. 
In my opinion, the first @JsonProperty is used for deserialization and this annotation is needed when you use @JsonCreator on constructor. but the second @JsonProperty is used to serialized.
We can't put only one annotation on the attribute, because @JsonCreator need to together with annotation @JsonProperty.
Last, this way is same like other json class, such as TableConfig (org.apache.pinot.spi.config.table.TableConfig)

Reply:
You can place `@JsonProperty` in the attribute. Alternatively, you can move the `@JsonPropertyDescription` from the attribute to the getter.

`@JsonProperty` can be placed in attributes, getters and setters (well, and attributes in `@JsonCreator`, but that the meaning there is different). In case an annotation is found in several places, the preference will be getter > setter > attribute. It doesn't mean that we should always place them in the getter, but it means that we should not place annotations in different places in order to make it more difficult to make mistakes.

My personal taste is to place them in attributes (as they are in the front of the class) but I'm not going to block a PR that adds them in the getter (especially if that is what we do in the rest of the code). But I really think that having annotations in attributes and in getters is error prone.

Reply:
@gortiz Thanks for your advice，I will try to